### PR TITLE
Feature/bed 13

### PR DESCRIPTION
**Pull request by ho-henry-senior**
**What changes does this PR bring?**

**Definition of Done**
The following need to be checked off as they are complete as part of the
PR process. If you are unable to mark off an item as complete, then state
in the comments why this is the case.
- [x] Have you built the code locally and confirmed its working?
- [x] Is the build confirmed to be passing on CI?
- [x] Have you added enough relevant unit tests for your change?
- [ ] Have you added enough relevant E2E tests for your change?
- [ ] Have you updated any relevant documentation as part of this change?
- [x] Has this change been tested against the Acceptance Criteria?
- [ ] Have you considered how this will be deployed in SIT/Staging/Production?

### CODE REVIEW

File: src/common/utils/validator.js [hunk 1]
```diff
@@ -217,18 +217,31 @@ function currentOrFutureDate(dObj) {
   return false;
 }
 
-// function dateTooFarInFuture(dObj){
-//   if (dObj === null || dObj === undefined) return false;
+function dateNotTooFarInFuture(dObj) {
```
Comment (anushasivakumar) [inline src/common/utils/validator.js:220]: Please can we add a docstring for what this function does

File: src/common/utils/validator.js [hunk 2]
```diff
@@ -217,18 +217,31 @@ function currentOrFutureDate(dObj) {
...
+
+  if (dObj instanceof Date) {
+    providedDate = dObj;
+  } else {
+
+    if (!(numericDateElements(dObj)
+      && validDay(dObj.d, dObj.m, dObj.y)
+      && validMonth(dObj.m)
+      && validYear(dObj.y))) {
+      return false;
```
Comment (anushasivakumar) [inline src/common/utils/validator.js:237]: Can we verify this check, please - currently this is returning false if it is not a numericvalue but a valid date/month/day

File: src/test/validators.test.js [hunk 3]
```diff
@@ -816,4 +817,44 @@ describe('Validator', () => {
...
+      expect(actualResult).to.equal(false);
+    });
+
+    it('Should accept a date within one month in the future in object format', () => {
+      actualResult = validator.dateNotTooFarInFuture({ d: '11', m: '5', y: '2023' });
+      expect(actualResult).to.equal(true);
+    });
+
+    it('Should reject a date further than one month in the future in date format', () => {
+      actualResult = validator.dateNotTooFarInFuture(new Date(2024, 0, 1));
```
Comment (anushasivakumar) [inline src/test/validators.test.js:847]: Is this test for an invalid date or for one month in the future in date format? The month seems to be '0'. Perhaps we need tests for both the cases.

**Review by anushasivakumar — COMMENTED** [review]
Overall looks good, a few minor comments added.

File: src/app/garfile/arrival/post.controller.js [hunk 4]
```diff
@@ -86,7 +86,7 @@ const buildValidations = (voyage) => {
   const validations = [
     [new ValidationRule(validator.realDate, 'arrivalDate', arriveDateObj, realDateMsg)],
     [new ValidationRule(validator.currentOrFutureDate, 'arrivalDate', arriveDateObj, futureDateMsg)],
-    // [new ValidationRule(validator.dateTooFarInFuture, 'arrivalDate', arriveDateObj, __('field_arrival_date_too_far_in_future'))],
+    // [new ValidationRule(validator.dateNotTooFarInFuture, 'arrivalDate', arriveDateObj, __('field_arrival_date_too_far_in_future'))],
```
[the remark this answers is not in the document]
Comment (ho-henry-senior) [reply src/app/garfile/arrival/post.controller.js:89]: ok will remove

File: src/common/utils/validator.js (hunk 1 above)
Comment (ho-henry-senior) [reply src/common/utils/validator.js:220]: will do

File: src/common/utils/validator.js (hunk 2 above)
Comment (anushasivakumar) [reply src/common/utils/validator.js:237]: It is good to have the check but we just need to double check if the condition is correct.

Comment (ho-henry-senior) [reply src/common/utils/validator.js:243]: have confirmed with Anusha that condition is correct

**Review by anushasivakumar — APPROVED** [review]

**Pull request merged**